test/examples_test.go

```diff
@@ -101,10 +128,34 @@ func TestHelloWorldOnChain(t *testing.T) {
 		Run("--grep", "Local")
 }
 
+func TestAttestFnCallCombined(t *testing.T) {
+	attestFnCallName := "attest_fn_call"
+	attestFnCallDir := filepath.Join(examplesDir, attestFnCallName)
+	helloWorldOnChainName := "hello_world_on_chain"
+	helloWorldOnChainDir := filepath.Join(examplesDir, helloWorldOnChainName)
+	saveDir := helloWorldOnChainDir + "/tmp"
+
+	t.Run(attestFnCallName, func(t *testing.T) {
+		NewTestscriptTest(t, attestFnCallDir, saveDir).
+			ExecuteMakeTarget("main.wasm").
+			CopyFile("main.wasm").
+			CopyFile("config.toml").
+			CopyFile("fn-call.json").
+			Run(filepath.Join(scriptDir, attestFnCallName+".txtar"))
+	})
+
+	t.Run(helloWorldOnChainName, func(t *testing.T) {
+		NewHardhatTest(t, helloWorldOnChainDir).
+			NPMInstall().
+			SetEnv("TA_FILE", "../tmp/attest-fn-call-out.json").
+			Run("--grep", "Local")
+	})
```

Comment:
As noted in another comment, you should be able to use t.SetEnv here, and the test framework will handle isolating/resetting the env values
```go
	t.Run(helloWorldOnChainName, func(t *testing.T) {
		// set the env directly in the test framework that's wrapping the hardhat test
		t.SetEnv("TA_FILE", "../tmp/attest-fn-call-out.json")
		
		NewHardhatTest(t, helloWorldOnChainDir).
			NPMInstall().
			Run("--grep", "Local")
	})
```

Reply:
Thanks, removed environment setup from `HardhatTest` and use `testing.T` now instead